amalgamation/amalgamation.py

```diff
@@ -143,7 +143,12 @@ def expand(x, pending, stage):
                 continue
             path = m.groups()[0]
             h = path.strip('./') if "../3rdparty/" not in path else path
-            source = find_source(h, x, stage)
+            if h.endswith('complex.h') and x.endswith('openblas_config.h'):
```

Comment:
Could you add a comment explaining the special case? Also, is there maybe a nicer way to deal with this ambiguity - e.g. making the names (complex.h) more unique in the output to avoid a name clash?